src/api/spec/factories/project.rb

```diff
@@ -77,11 +77,21 @@
 
       transient do
         target_project nil
+        create_patchinfo false
+        maintainer nil
       end
 
       after(:create) do |project, evaluator|
         create(:maintainance_project_attrib, project: project)
-        create(:maintained_project, project: evaluator.target_project, maintenance_project: project) if evaluator.target_project
+        if evaluator.target_project
+          create(:maintained_project, project: evaluator.target_project, maintenance_project: project)
+          CONFIG['global_write_through'] ? project.store : project.save!
+        end
+        if evaluator.create_patchinfo
```

Comment:
why not using here `if evaluator.maintainer` and you can delete the `create_patchinfo` variable?